### ISSUE TITLE
Add Support for Dynamic Templates

### ISSUE THREAD

**Issue by richardherbert** [mjclemente/sendgrid.cfc#5]

**Comment by mjclemente** [comment]
Thanks. Definitely worth adding.

Found the docs for the Dynamic Templates: https://sendgrid.com/docs/ui/sending-email/how-to-send-an-email-with-dynamic-transactional-templates/

Just putting that here for reference.

**Comment by mjclemente** [comment]
Support added.

Closed with 8f85cc443af1ff8958ed850dcb3e5616cb0190d0

Example Usage, using [SendGrid's example receipt template](https://github.com/sendgrid/email-templates/blob/master/dynamic-templates/receipt/receipt.html).

```cfc
  sg = new path.to.sendgridcfc.sendgrid();

  fromEmail = 'hello@youremail.com';
  toEmail = 'someone@else.com';
  templateId = 'd-put-the-dynamic-template-id-here';
  data = {
    "total": "$ 239.85",
    "items": [
      {
        "text": "New Line Sneakers",
        "image": "https://marketing-image-production.s3.amazonaws.com/uploads/8dda1131320a6d978b515cc04ed479df259a458d5d45d58b6b381cae0bf9588113e80ef912f69e8c4cc1ef1a0297e8eefdb7b270064cc046b79a44e21b811802.png",
        "price": "$ 79.95"
      }
    ],
    "receipt": true,
    "name": "Sample Name",
    "address01": "1234 Fake St.",
    "address02": "Apt. 123",
    "city": "Place",
    "state": "CO",
    "zip": "80202"
  };

  mail = new path.to.sendgridcfc.helpers.mail()
  .from( fromEmail )
  .to( toEmail )
  .templateId( templateId )
  .withDynamicTemplateData( data );
```

**Comment by richardherbert** [comment]
Thanks for taking that on Matthew. I'll take a look at migrating my template over.
